activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -260,7 +260,17 @@ def include?(record)
           if record.new_record?
             include_in_memory?(record)
           else
-            loaded? ? target.include?(record) : scope.exists?(record.id)
+            if loaded?
+              target.include?(record)
+            else
+              id = if record.class.composite_primary_key?
+                record.class.primary_key.zip(record.id).to_h
+              else
+                record.id
+              end
+
+              scope.exists?(id)
+            end
```

Comment:
Trivial: Probably, we can use this opportunity to refactor the code a little bit for better readability. Something like this: 

```ruby
def include?(record)
  klass = reflection.klass
  return false unless record.is_a?(klass)

  if record.new_record?
    include_in_memory?(record)
  elsif loaded?
    target.include?(record)
  else
    record_id = klass.composite_primary_key? ? klass.primary_key.zip(record.id).to_h : record.id
    scope.exists?(record_id)
  end
end
```

Reply:
Thank you, that indeed looks cleaner!